File: server/Infrastructure/LobTools/Models/History/EntityTypeHistoryModel.cs
namespace Brainvest.Dscribe.LobTools.Models;

using System;
using Brainvest.Dscribe.Abstractions;
using Brainvest.Dscribe.Abstractions.Models.ManageMetadata;

public class EntityTypeHistoryModel : IHistory
{
	public EntityTypeModel EntityType { get; set; }
	public DateTime StartTime { get; set; }
	public Guid? UserId { get; set; }
	public double ProcessDuration { get; set; }
	public long LogId { get; set; }
	public string Action { get; set; }
}

File: client/src/lib/administration/models/history/entity-type-history-model.ts
import { IHistory } from './history-model';
import { HistoryType } from './history-type';
import {EntityTypeBase} from '../../../metadata/entity-type-base';
import {MetadataBasicInfoModel} from '../../../metadata/metadata-basic-info-model';

export class EntityTypeHistoryModel implements IHistory {
	EntityType: EntityTypeBase;
	LogId: number;
	ProcessDuration: string;
	Action: string;
	StartTime: Date;
	UserId: string;
	basicInfo: MetadataBasicInfoModel;
	historyType: HistoryType;

	constructor() {
		this.EntityType = new EntityTypeBase();
		this.EntityType.EntityTypeGeneralUsageCategoryId = 1;
		this.EntityType.Name = 'a';
		this.EntityType.PluralTitle = 'a';
		this.EntityType.SingularTitle = 'a';
	}
}


